tests/test_models/test_tbats.py

```diff
@@ -127,3 +170,168 @@ def test_prediction_interval(model, example_tsds):
 @pytest.mark.parametrize("model", [TBATSModel(), BATSModel()])
 def test_save_load(model, example_tsds):
     assert_model_equals_loaded_original(model=model, ts=example_tsds, transforms=[], horizon=3)
+
+
+def test_forecast_decompose_not_fitted(small_periodic_ts):
+    model = _TBATSAdapter(model=BATS())
+
+    with pytest.raises(ValueError, match="Model is not fitted!"):
+        model.forecast_components(df=small_periodic_ts.df)
+
+
+def test_predict_components_not_implemented(small_periodic_ts):
+    model = _TBATSAdapter(model=BATS())
+
+    with pytest.raises(NotImplementedError, match="Prediction decomposition isn't currently implemented!"):
+        model.predict_components(df=small_periodic_ts.df)
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_decompose_forecast_output_format(small_periodic_ts, estimator):
+    horizon = 3
+    model = estimator()
+    model.fit(small_periodic_ts)
+
+    components = model._models["segment_1"]._decompose_forecast(horizon=horizon)
+    assert isinstance(components, np.ndarray)
+    assert components.shape[0] == horizon
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_named_components_output_format(small_periodic_ts, estimator):
+    horizon = 3
+    model = estimator()
+    model.fit(small_periodic_ts)
+
+    segment_model = model._models["segment_1"]
+    components = segment_model._decompose_forecast(horizon=horizon)
+    components = segment_model._process_components(raw_components=components)
+
+    assert isinstance(components, pd.DataFrame)
+    assert len(components) == horizon
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator,params,components_names",
+    (
+        (
+            BATSModel,
+            {"use_box_cox": False, "use_trend": True, "use_arma_errors": True, "seasonal_periods": [7, 14]},
+            {"local_level", "trend", "arma(p=1,q=1)", "seasonal(s=7)", "seasonal(s=14)"},
+        ),
+        (
+            TBATSModel,
+            {"use_box_cox": False, "use_trend": True, "use_arma_errors": False, "seasonal_periods": [7, 14]},
+            {"local_level", "trend", "seasonal(s=7.0)", "seasonal(s=14.0)"},
+        ),
+    ),
+)
+def test_components_names(periodic_ts, estimator, params, components_names):
+    train, test = periodic_ts
+    model = estimator(**params)
+    model.fit(train)
+
+    future = train.make_future(3).to_pandas(flatten=True)
+
+    for segment in test.columns.get_level_values("segment"):
+        components_df = model._models[segment].forecast_components(df=future)
+        assert set(components_df.columns) == components_names
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_seasonal_components_not_fitted(small_periodic_ts, estimator):
```

Comment:
May you also test the combination of components arma + seasonal if it is a possible scenario

Reply:
Added